cli/replicaset/cmd/upgrade.go

```diff
@@ -116,33 +117,53 @@ func getInstanceConnector(instance replicaset.Instance) (connector.Connector, er
 	})
 
 	if err != nil {
-		return nil, fmt.Errorf("instance %s failed to connect via UNIX socket "+
-			": %w", fullInstanceName, err)
+		fErr := err
+		conn, err = connector.Connect(connector.ConnectOpts{
+			Network:  connOpts.Network,
+			Address:  instance.URI,
+			Username: connOpts.Username,
+			Password: connOpts.Password,
+			Ssl:      connOpts.Ssl,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("instance %s failed to connect via UNIX socket "+
+				"and uri: %w %w", fullInstanceName, err, fErr)
+		}
 	}
 	return conn, nil
 }
 
-func collectRWROInfo(replset replicaset.Replicaset) (*instanceMeta, []instanceMeta,
+func collectRWROInfo(replset replicaset.Replicaset,
+	connOpts connector.ConnectOpts) (*instanceMeta, []instanceMeta,
 	error) {
 	var master *instanceMeta = nil
 	var replicas []instanceMeta
 	for _, instance := range replset.Instances {
 		run := instance.InstanceCtx
 		fullInstanceName := running.GetAppInstanceName(run)
-		conn, err := getInstanceConnector(instance)
+		conn, err := getInstanceConnector(instance, connOpts)
 
 		if err != nil {
 			return nil, nil, err
 		}
 
+		isRW := false
+
 		if instance.Mode == replicaset.ModeUnknown {
```

Comment:
This point was already [discussed](https://github.com/tarantool/tt/pull/936#discussion_r1817027430) (in the previous patch), but I found that the Discovery mechanism cannot determine the mode of all instances in a replica set (although it should) using the uri.

For example, I try to run `tt replicates status` to demonstrate the problem:
```bash
$ tt replicaset status tcp://client:secret@127.0.0.1:3301
Orchestrator:      centralized config
Replicasets state: bootstrapped

• storage-001
  Failover: manual
    • storage-001-a localhost:3301 rw
    • storage-001-b localhost:3302 unknown
```
But it knows the correct address `localhost:3302` to connect to this instance. I think this is a bug. Without the uri everything is fine:
```bash
$ tt replicaset status new-app:storage-001-a
Orchestrator:      centralized config
Replicasets state: bootstrapped

• router-001
  Failover: off
  Master:   single
    • router-001-a localhost:3305 rw
• storage-001
  Failover: manual
  Master:   single
    • storage-001-a localhost:3301 rw
    • storage-001-b localhost:3302 read
• storage-002
  Failover: manual
  Master:   single
    • storage-002-a localhost:3303 rw
    • storage-002-b localhost:3304 read
```